src/porepy/numerics/ad/grid_operators.py

```diff
@@ -87,12 +87,13 @@ def cell_restriction(self, grids: List[pp.Grid]) -> Matrix:
             if len(grids) > 0:
                 # A key error will be raised if a grid in g is not known to
                 # self._cell_projection
+                # IMPLEMENTATION NOTE: Use csc format, since the number of rows can
+                # be much less than the number of columns.
                 mat = sps.bmat([[self._cell_projection[g].T] for g in grids]).tocsr()
```

Comment:
Why not use the new function here?

Reply:
Because explicit is better than implicit. In this case the shape of the matrix is clear, it is a single matrix (no product, thus clean code) and the format can be set directly.  For the cases further up in the file, the size of the matrix can be inferred, but it requires reasoning, and calling the function overall gives the best readability. In the mortar grid class, the size cannot be inferred in the case of non-matching grids, and the function is by far the better option.